easybuild/easyblocks/c/code_saturne.py

```diff
@@ -0,0 +1,158 @@
+##
+# Copyright 2009-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Code_Saturne, implemented as an easyblock
+
+@author: Metin Cakircali (Juelich Supercomputing Centre)
+"""
+
+import os
+
+import easybuild.tools.environment as env
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.run import run_cmd
+from easybuild.tools.filetools import apply_regex_substitutions
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_Code_underscore_Saturne(EasyBlock):
+    """Support for building and installing Code_Saturne."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to Code_Saturne."""
+        extra_vars = EasyBlock.extra_options(extra_vars)
+
+        # add more custom easyconfig parameters specific to OpenFOAM
+        extra_vars.update({
+            'debug': [False, "Build the debug version.", CUSTOM],
+            'slurm': [False, "Build for the slurm resource manager.", CUSTOM],
+        })
+
+        return extra_vars
+
+    def prepare_step(self, *args, **kwargs):
+        """Prepare step for Code_Saturne obtained from the repository."""
+        super(EB_Code_underscore_Saturne, self).prepare_step(*args, **kwargs)
+
+        self.log.info("Running ./sbin/bootstrap ...")
+
+        cmd = './sbin/bootstrap'
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
```

Comment:
@mcakircali The tests fail right now because `prepare_step` is also run when `eb --module-only` is used, but the build directory is not set up under `--module-only` (so there's no unpacking, etc.).

You can fix this in two ways, up to you which makes more sense to you:

* move running `./sbin/bootstrap` to `configure_step` (which is skipped under `--module-only`)
* only conditionally run `./sbin/bootstrap`, e.g. when the file is there, or when `--module-only` is not used (you can check the latter with `if build_option('module_only'):`)

Reply:
I like option 2

Reply:
@boegel thanks!
I like option 1, as it's more logical. there's no meaning to do bootstrap step for --module-only.. also option 2 sounds like a workaround that complicates things.